OpenRA.Mods.Common/Widgets/Logic/Editor/SaveMapLogic.cs

```diff
@@ -191,19 +191,25 @@ public SaveDirectory(Folder folder, MapClassification classification)
 					}
 
 					map.Save(package);
-				}
-				catch
-				{
-					Console.WriteLine("Failed to save map at {0}", combinedPath);
-				}
 
-				// Update the map cache so it can be loaded without restarting the game
-				modData.MapCache[map.Uid].UpdateFromMap(map.Package, selectedDirectory.Folder, selectedDirectory.Classification, null, map.Grid.Type);
+					// Update the map cache so it can be loaded without restarting the game
+					modData.MapCache[map.Uid].UpdateFromMap(map.Package, selectedDirectory.Folder, selectedDirectory.Classification, null, map.Grid.Type);
 
-				Console.WriteLine("Saved current map at {0}", combinedPath);
-				Ui.CloseWindow();
+					Console.WriteLine("Saved current map at {0}", combinedPath);
+					Ui.CloseWindow();
 
-				onSave(map.Uid);
+					onSave(map.Uid);
+				}
+				catch (Exception e)
+				{
+					Console.WriteLine("Failed to save map at {0}: {1}", combinedPath, e.Message);
```

Comment:
I think the best way to do this would be to add `Log.Write("debug", "Failed to save map at {0}: {1}", combinedPath, e.Message);` and then change the dialog text to "See debug.log for details".  This avoids duplication in the UI message and directs map authors to where they can find out what went wrong.